### ISSUE TITLE
easy to remeber but secrue password - modify password gen plugin

### ISSUE THREAD

**Issue by Abdur-rahmaanJ** [Abdur-rahmaanJ/honeybot#41]
like  i-lov3@swimming4lif3
find replacement for normal english letters

**Comment by Abdur-rahmaanJ** [comment]
no, just add another command to the password plugin

**Comment by msarfrazanwar** [comment]
> no, just add another command to the password plugin

I need more clarification on the requirements as I am not getting the exact point. Give some more examples if possible with user input and password generation.

**Comment by Abdur-rahmaanJ** [comment]
@msarfrazanwar just add another command here [https://github.com/Abdur-rahmaanJ/honeybot/blob/master/honeybot/plugins/password_generator.py](https://github.com/Abdur-rahmaanJ/honeybot/blob/master/honeybot/plugins/password_generator.py)

**Comment by msarfrazanwar** [comment]
I know. What kind of passwords to be generated can you give some examples ?? We are not working on ML or something that I generate sentences like I love swimming for life.

**Comment by Abdur-rahmaanJ** [comment]
@msarfrazanwar see username plugin. something like

```
p = ['I', 'You', 'He', 'She']
verbs = [ # depends on i or he
    [['love'], ['loves']],
    [['like'], ['likes']],
    [['hate'], ['hates']],
]
actions = ['walking', 'climbing', 'rolling']
```

**Comment by msarfrazanwar** [comment]
Yes that is what I was asking about. Thanks. I will try my best.

**Comment by Abdur-rahmaanJ** [comment]
👍

**Comment by msarfrazanwar** [comment]
Will there be any input to the function like number of words or anything like that?

**Comment by Abdur-rahmaanJ** [comment]
no, you devise something reasonable

>

**Comment by Macr0Nerd** [comment]
But in using a set list of verbs and such, it becomes that much easier to break the security of the plugin. I propose both easy and secure password options. Such as one that generates the example above, but also one that puts together a random string of characters.

**Comment by msarfrazanwar** [comment]
@Abdur-rahmaanJ @Macr0Nerd  Currently my generating passwords like this : 
-her silly screwdriver might be beyond her blurry
-the only typical computer program will enjoy towards the only saturated t-shirt
-our helpless project is thinking unlike every strange beach
I can the length and add special characters for security. What do you guys think ?

**Comment by Macr0Nerd** [comment]
@msarfrazanwar That seems like a smart approach. I'd say just replace some of the letters with characters akin to 1337 Speak, so you get stuff like H3r$1lly$cr3wd1v3rM1ghtB3B3y0ndH3rBlurry. I was proposing a Random Password Generator such as:
    import random

    char = ['A', 'b', '@', '9'] #so on with all the upper and lower case characters, numbers, and special chars

    pass = ''
    for i in range(16):
        pass += random.choice(char)

    return pass

If we really felt it necessary, we could also throw in an unsalted hash. It might also be worthwhile to look into a secure password generator.

**Comment by Abdur-rahmaanJ** [comment]
random words only not great for sec, yes add special chars in place of
letters.

**Comment by msarfrazanwar** [comment]
Hey, I have developed the generator. I am writing down the type of passwords it generating. What do you guys think about this?
-h3r b1urry $@x0ph0n3
-h1$ gr@y pr0j3ct
-h1$ br@v3 t-$h1rt
-th@t $@tur@t3d h@rd dr1v3
-on3 un1qu3 c@rp3t

**Comment by Abdur-rahmaanJ** [comment]
seems great!

**Comment by Macr0Nerd** [comment]
Do we want the white space? That’s the only qualm I have with it, otherwise it looks awesome

**Comment by Abdur-rahmaanJ** [comment]
yes without spaces better

**Comment by Macr0Nerd** [comment]
@msarfrazanwar @Abdur-rahmaanJ I mentioned the wrong issue in PR #59. After some testing, this plugin did not seem to work.

**Comment by Abdur-rahmaanJ** [comment]
@msarfrazanwar PR

**Comment by msarfrazanwar** [comment]
@Macr0Nerd I have the body of function which will return the secure password can you with where it will be called in that file?

**Comment by Macr0Nerd** [comment]
@msarfrazanwar looks like it should work. I'll do some testing within the next day and just make sure

**Comment by msarfrazanwar** [comment]
@Macr0Nerd Is there any medium where we can communicate in real time ?. I need to discuss a small thing.

**Comment by Macr0Nerd** [comment]
@msarfrazanwar I’m now in the discord. It should help to streamline things.

**Comment by msarfrazanwar** [comment]
@Macr0Nerd Can you provide me a link or something to join you there?

**Comment by Abdur-rahmaanJ** [comment]
see bottom of readme ^^_

**Comment by Macr0Nerd** [comment]
@Abdur-rahmaanJ @msarfrazanwar what's the status of the issue?

**Comment by msarfrazanwar** [comment]
@Macr0Nerd I password generator function is all ready and working but I am not able to understand how to integrate in that python file with other commands.

**Comment by vetragor** [comment]
how to join the project

**Comment by Abdur-rahmaanJ** [comment]
@vetragor see discord link in readme

**Comment by WillBurnham** [comment]
Hi, I created a solution that will randomly generate a password like:
1dr1nkund3racomputer
Sh3g03st0th3b3d
h3g03sinastore
h3runs@l0ng@c@r
etc. 
It essentially uses lists to form sentences and then a hash table to change normal characters to 1337 speak. I programmed it to only 1337 speak half of the password on occasion to help with predictability. If this issue is still open and this sounds like it is the right solution, could I work on this?

**Comment by Abdur-rahmaanJ** [comment]
@WillBurnham yes just it's here https://github.com/pyhoneybot/pyhoneybot
